test/hotspot/jtreg/runtime/cds/appcds/aotCache/JavaAgentTransformer.java

```diff
@@ -30,11 +31,14 @@
 // This class is available on the classpath so it can be accessed by JavaAgentApp
 public class JavaAgentTransformer  implements ClassFileTransformer {
     private static Instrumentation savedInstrumentation;
+    private static final System.Logger LOGGER = System.getLogger(JavaAgentTransformer.class.getName());
 
     public static void premain(String agentArguments, Instrumentation instrumentation) {
         System.out.println("JavaAgentTransformer.premain() is called");
         instrumentation.addTransformer(new JavaAgentTransformer(), /*canRetransform=*/true);
         savedInstrumentation = instrumentation;
+
+        LOGGER.log(Level.WARNING, "JavaAgentTransformer::premain() is finished");
```

Comment:
Just curious why not use `Level.INFO` instead. Is it because the log won't be printed with product build if `Level.INFO` is used?

Reply:
I copied this code from the original reproducer. If this code was executed inside the assembly phase, then we will get a CDSHeapVerifier error.